enhancements/bare-metal-fulfillment/README.md

```diff
@@ -0,0 +1,360 @@
+
+#### ESI Playbooks
+
+The ESI API supports all of the needed bare metal functionality (the cluster fulfillment workflow already relies upon ESI
+playbooks for bare metal network configuration) . However, it is important to recognize that O-SAC will not use the ESI concept
+of tenancy; from an ESI perspective, hosts will always remain leased to the O-SAC project. As a result, we will need to introduce
+a new method of keeping track of O-SAC tenancy.
+
+One solution would be to rely upon Ironic’s extra field, which allows an ESI project to set arbitrary key/value pairs upon a
+specified host. Doing so would not be difficult; it simply creates additional considerations when implementing playbooks related
+to host assignment and unassignment.
+
+#### Serial Console Support
+
+There are multiple directions we could explore for serial console support:
+
+* ESI already supports serial console access through socat (using an Ironic feature exposed through an ESI proxy). We could simply
+generate the direct console URL and store that in the resource status.
+* We could implement serial console support similar to the pod exec api (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#execaction-v1-core)
+
+The first solution has the advantage of already being implemented within ESI; all we have to do is expose the URL generated by
+ESI. However, it is unknown whether other bare metal services would have similar support for serial console access; for that
+reason, we may want to investigate the possibility of a more generic solution that can be consistently used across multiple
+bare metal management implementations.
+
+#### Task Groupings
+
+We can perform this implementation in smaller steps:
+
+* HostPool and Host
+  * Tenants can create/update/delete HostPools
+  * No network attachments
+  * Hosts are created when a HostPool is fulfilled
+  * Tenants can view their Hosts
+  * Tenants have power control on their Hosts
+* Implement HostPool hostSelectors
+  * filter by name (to support removal of specific hosts from a HostPool)
+  * filter by rack
+  * affinity/anti-affinity
+* Network Attachments
+  * Tenants can specify network attachments when creating/updating HostPools
+  * Requires network service
+* Serial Console
+  * Tenants can enable/disable serial console on a Host
+
+### Risks and Mitigations
+
+??
+
+### Drawbacks
+
+??
+
+## Alternatives (Not Implemented)
+
+One alternative would be to focus on an ESI replacement before developing the bare metal fulfillment
+functionality. However, we anticipate that the MOC will continue to use ESI for now; for that reason,
+it makes sense to perform this implementation with ESI in mind, and simply make it easy to replace
+ESI if/when needed. Note that it is also possible that we will continue to use ESI if the issues
+pushing us towards a replacement are fixed.
+
+## Open Questions [optional]
+
+* Can we remove network attachment by MAC address?
+
```

Comment:
What about attaching the same network to multiple interfaces? Supported? Not supported?